### PR TITLE
MEGA Toys presents: Xenos Isolated action figure!

### PR DESCRIPTION
**Pull request by ghost**
MEGA proudly presents the new xenomorph actions figure! Comes complete with realistic sounds that can be activated by pulling back string trigger. Can be found in any normal arcade machine.

Demonstration: (https://dl.dropboxusercontent.com/u/116194515/toy_xeno.mp4)

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1171,4 +1172,32 @@
...
+	if(cooldown < world.time)
+		cooldown = (world.time + 50) //5 second cooldown
+		user.visible_message("<span class='notice'>[user] pulls back the string on [src].</span>")
+		icon_state = "[initial(icon_state)]_used"
+		sleep(5)
+		audible_message("<span class='danger'>\icon[src] Hiss!</span>")
+		var/list/possible_sounds = list('sound/voice/hiss1.ogg', 'sound/voice/hiss2.ogg', 'sound/voice/hiss3.ogg', 'sound/voice/hiss4.ogg')
+		var/chosen_sound = pick(possible_sounds)
+		playsound(get_turf(src), chosen_sound, 50, 1)
+		spawn(45)
```
Comment (RemieRichards) [inline code/game/objects/items/toys.dm]: this spawn is empty, and does nothing.

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1171,4 +1172,32 @@
...
+/obj/item/toy/toy_xeno
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "toy_xeno"
+	name = "xenomorph action figure"
+	desc = "MEGA presents the new Xenos Isolated action figure! Comes complete with realistic sounds! Pull back string to use."
+	w_class = 2
+	var/cooldown = 0
+
+/obj/item/toy/toy_xeno/attack_self(mob/user)
+	if(cooldown < world.time)
```
Comment (RemieRichards) [inline code/game/objects/items/toys.dm]: ```<=```

File: code/game/objects/items/toys.dm (hunk 1 above)
Comment (Xhuis) [inline code/game/objects/items/toys.dm]: fixed

Comment (RemieRichards) [inline code/game/objects/items/toys.dm]: Not quite, you never checked the object still exists after the spawn

**Comment by WJohn** [comment]
:+1: x :100: 
Oh my god I want one.

**Comment by Jalleo** [comment]
Image preferably a video though please.

**Comment by Xhuis** [comment]
OP updated with demonstration

**Comment by WJohn** [comment]
I don't think the string on the sprite looks very good, and I'm not sure it even needs one. The delay between uses is very short anyway, so why have an extra sprite for it at all?

**Comment by Xhuis** [comment]
Figured it'd add some extra flavor. I'm a coder, not a spriter.

**Comment by Nienhaus** [comment]
I have this if you want to code it.
https://www.dropbox.com/s/9wcpypentioaawb/209%20toy.dmi?dl=0

**Comment by Supermichael777** [comment]
Coming soon: woody got wood action-figure, with real bulging action!

**Comment by Shadowlight213** [comment]
Some server had a woody doll that ran around gibbing people.

**Comment by duncathan** [comment]
NoX had only the most important features

**Comment by Miauw62** [comment]
dehydrated xenos when

**Comment by Jalleo** [comment]
You mean dehydrated clockwork xenos

**Comment by Shapsy** [comment]
There should be a 1/100 chance of doing the deathscream

**Comment by Xhuis** [comment]
merge when

**Comment by Xhuis** [comment]
:alien: ayy lmao :alien:

**Pull request merged**